WooCommerce/Classes/ViewRelated/NotificationsBadgeController.swift

```diff
@@ -0,0 +1,102 @@
+import Foundation
+import UIKit
+
+final class NotificationsBadgeController {
+    /// Displays or Hides the Dot, depending on the new Badge Value
+    ///
+    func badgeCountWasUpdated(newValue: Int, in tabBar: UITabBar) {
+        guard newValue > 0 else {
+            hideDotOn(.notifications, in: tabBar)
+            return
+        }
+
+        showDotOn(.notifications, in: tabBar)
+    }
+
+    /// Shows the dot in the specified WooTab
+    ///
+    func showDotOn(_ tab: WooTab, in tabBar: UITabBar) {
+        hideDotOn(tab, in: tabBar)
+        let dot = GreenDotView(frame: CGRect(x: DotConstants.xOffset,
+                                             y: DotConstants.yOffset,
+                                             width: DotConstants.diameter,
+                                             height: DotConstants.diameter), borderWidth: DotConstants.borderWidth)
+        dot.tag = dotTag(for: tab)
+        dot.isHidden = true
+        tabBar.subviews[tab.rawValue].subviews.first?.insertSubview(dot, at: 1)
+        dot.fadeIn()
+    }
+
+    /// Hides the Dot in the specified WooTab
+    ///
+    func hideDotOn(_ tab: WooTab, in tabBar: UITabBar) {
+        let tag = dotTag(for: tab)
+        if let subviews = tabBar.subviews[tab.rawValue].subviews.first?.subviews {
+            for subview in subviews where subview.tag == tag {
+                subview.fadeOut() { _ in
+                    subview.removeFromSuperview()
+                }
+            }
+        }
+    }
+
+    /// Returns the DotView's Tag for the specified WooTab
+    ///
+    func dotTag(for tab: WooTab) -> Int {
+        return tab.rawValue + DotConstants.tagOffset
+    }
+}
+
+
+// MARK: - Constants!
+//
+private extension NotificationsBadgeController {
+
+    enum DotConstants {
+        static let diameter    = CGFloat(10)
```

Comment:
Figma file shows the dot as 15pts. I'm not exactly sure how that translates here, it might be `CGFloat(15)` or it might be `CGFloat(7.5)`

Reply:
I've updated this in PR #1152